gui.py

```diff
@@ -159,10 +164,30 @@ def __code_generieren(self):
         """
         Startet den Prozess der Codegenerierung
         """
+        """
+        Codegenerierung ohne interaktive Eingabe der Kontaktdaten
 
-        # TODO: code generierung implementieren
-        QtWidgets.QMessageBox.information(self, "Noch nicht verfügbar", "Funktion nur über Konsolenanwendung verfügbar")
+        :param kontaktdaten: Dictionary mit Kontaktdaten
+        """
+        
+        #disable button after click
+        self.b_code_generieren.setEnabled(False) 
+        try:
+            kontaktdaten = self.__get_kontaktdaten(Modus.CODE_GENERIEREN)
 
+        except FileNotFoundError as error:
+            QtWidgets.QMessageBox.critical(self, "Datei nicht gefunden!", f"Datei zum Laden konnte nicht gefunden werden\n\nBitte erstellen")
+            return
+        except ValidationError as error:
+            QtWidgets.QMessageBox.critical(self, "Daten Fehlerhaft!", f"In der angegebenen Datei sind Fehler:\n\n{error}")
+            return
+        except MissingValuesError as error:
+            QtWidgets.QMessageBox.critical(self, "Daten Fehlerhaft!", f"In der angegebenen Datei Fehlen Daten:\n\n{error}")
+            return
+            
+    
+        dialog = QtCodeGen(self, kontaktdaten, PATH )
```

Comment:
Vorschlag dass auch wirklich alles Beendet wird, nach dem ein Code generiert wurde.
Müsstest dann noch entsprechend dem Beispiel hier eine `@staticmethod start_code_gen` implementieren
`self` ist hier wichtig, damit der Prozess nicht durch die eigene Lebenszeit in der Methode stirbt
```suggestion
                self.terminsuche_prozess = multiprocessing.Process(target=QtCodeGen.start_code_gen, name=f"code_gen", daemon=True, kwargs={
                                                      "kontaktdaten": kontaktdaten,
                                                      "ROOT_PATH": PATH})
        self.terminsuche_prozess.start()
```

Reply:
Muss ich mir nochmal anschauen, aber das lagert ja die GUI weider in ein komplett eigenen Prozess aus nicht?

Reply:
ja genau, dadurch das die Threads ja blöderweise nicht beendet werden können, sorgt deshalb das beenden des Prozesses, dass auch direkt die Threads beendet werden, aber du hast das ja jetzt über Signale gelöst.
Wenn das funktioniert, dann mach dass lieber so :)